### PR TITLE
LobbyQuery Numerical and NearVal Filters, SteamClient and Friend OwnsLobby

### PR DESCRIPTION
**Pull request by thesupersoup**
Added LobbyQuery numerical filters for WithLower, WithHigher, WithEqual, and WithNotEqual; added OrderByNear for near value filters. Added OwnsLobby method to SteamClient and Friend which accepts a Lobby argument to test for ownership (replacing what I've been using: "if(SteamClient.SteamId == lobby.Owner.Id)")

### CODE REVIEW

File: Facepunch.Steamworks/SteamClient.cs [hunk 1]
```diff
@@ -169,6 +170,11 @@ internal static void UnregisterCallback( IntPtr intPtr )
 		/// </summary>
 		public static AppId AppId { get; internal set; }
 
+		/// <summary>
+		/// Check if this SteamClient owns the specified lobby
+		/// </summary>
+		public static bool OwnsLobby( Lobby k ) => k.Owner.Id == SteamId;
```
Comment (garrynewman) [inline Facepunch.Steamworks/SteamClient.cs]: This and the one in friends is really the wrong way around. You'd add this function to the Lobby if anywhere.

File: Facepunch.Steamworks/Structs/LobbyQuery.cs [hunk 2]
```diff
@@ -67,6 +63,87 @@ public LobbyQuery WithKeyValue( string key, string value )
 		}
 		#endregion
 
+		#region Numerical filters
+		internal Dictionary<KeyValuePair<string, int>, LobbyComparison> numericalFilters;
```
Comment (garrynewman) [inline Facepunch.Steamworks/Structs/LobbyQuery.cs]: I don't think Dictionary is the right way to do this. A List with a custom struct would make more sense.

**Comment by thesupersoup** [comment]
Made the recommended changes: added NumericalFilter struct and changed dict to list, changed ownership test to IsOwnedBy in Lobby

File: Facepunch.Steamworks/SteamClient.cs (hunk 1 above)
Comment (thesupersoup) [reply Facepunch.Steamworks/SteamClient.cs]: Ah, I do see that now. Thank you.

**Comment by garrynewman** [comment]
Looks good - thanks!

**Pull request merged**